frigate/log.py

```diff
@@ -1,71 +1,69 @@
-# adapted from https://medium.com/@jonathonbao/python3-logging-with-multiprocessing-f51f460b8778
+import atexit
 import logging
 import multiprocessing as mp
 import os
-import queue
-import signal
 import threading
 from collections import deque
-from logging import handlers
-from multiprocessing import Queue
-from types import FrameType
+from contextlib import AbstractContextManager, ContextDecorator
+from logging.handlers import QueueHandler, QueueListener
+from types import TracebackType
 from typing import Deque, Optional
 
-from setproctitle import setproctitle
+from typing_extensions import Self
 
 from frigate.util.builtin import clean_camera_user_pass
 
+LOG_HANDLER = logging.StreamHandler()
+LOG_HANDLER.setFormatter(
+    logging.Formatter(
+        "[%(asctime)s] %(name)-30s %(levelname)-8s: %(message)s",
+        "%Y-%m-%d %H:%M:%S",
+    )
+)
 
-def listener_configurer() -> None:
-    root = logging.getLogger()
+LOG_HANDLER.addFilter(
+    lambda record: not record.msg.startswith("You are using a scalar distance function")
+)
 
-    if root.hasHandlers():
-        root.handlers.clear()
 
-    console_handler = logging.StreamHandler()
-    formatter = logging.Formatter(
-        "[%(asctime)s] %(name)-30s %(levelname)-8s: %(message)s", "%Y-%m-%d %H:%M:%S"
-    )
-    console_handler.setFormatter(formatter)
-    root.addHandler(console_handler)
-    root.setLevel(logging.INFO)
+class log_thread(AbstractContextManager, ContextDecorator):
+    def __init__(self, *, handler: logging.Handler = LOG_HANDLER):
+        super().__init__()
 
+        self._handler = handler
```

Comment:
```suggestion
        logging.getLogger().handlers.clear()
        logging.getLogger().addHandler(handler)
        self._handler = handler
```

Reply:
this seems to take care of that issue

Reply:
I'm reluctant to do that because of the locks the logger normally takes to touch the handler list. It _should_ be fine given that this is normally called early on, but I'd still rather not risk it.

Let me investigate a bit.

Reply:
From the [Python docs for basicConfig()](https://docs.python.org/3/library/logging.html#logging.basicConfig):
> This function does nothing if the root logger already has handlers configured, unless the keyword argument force is set to True.

We can probably work around this by setting handlers to `[NullHandler()]` in `__main__` for now. I'll also send a PR to onvif to remove the `basicConfig()` call while I'm at it...

Reply:
I feel so dumb for noticing the double logs...
Anyway, this should be fixed: `handlers=[NullHandler()]` seems to work great for this.